File: Trees/LC_226.cpp
// Written by: Jian-Yuan Lin
// Date: 2022.11.29

class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        return invert(root);   
    }
private:
    TreeNode* invert(TreeNode* root){
        if (root == nullptr){
            return root;
        }
        if (root->left == nullptr && root->right == nullptr){
            return root;
        }
        swap(root->left, root->right);
        invert(root->left);
        invert(root->right);
        return root;
    }
};

File: Trees/LC_226.py
# Written by: Jian-Yuan Lin
# Date: 2022.11.29

class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        def invert(root):
            if not root:
                return root
            root.left, root.right = invert(root.right), invert(root.left)
            return root
        return invert(root)